src/main/scala/chisel3/util/LTL.scala

```diff
@@ -0,0 +1,392 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util.ltl
+
+import chisel3._
+import chisel3.util.circt._
+import chisel3.experimental.hierarchy.{Instance, Instantiate}
+
+/** An opaque sequence returned by an intrinsic.
+  *
+  * Due to the lack of opaque user-defined types in FIRRTL, the LTL intrinsics
+  * operate on `Bool` values instead of a more desirable `Sequence` type. To
+  * prevent abuse, the Chisel API here wraps these `Bool`s in an opaque type
+  * over which we can provide a safe interface that cannot be used to construct
+  * invalid IR.
+  */
+case class OpaqueSequence(private[ltl] val inner: Bool) extends Sequence
+
+/** An opaque property returned by an intrinsic.
+  *
+  * See `OpaqueSequence` for details.
+  */
+case class OpaqueProperty(private[ltl] val inner: Bool) extends Property
```

Comment:
This doc is interesting and all, but do these really need to be part of the public API? These could be private and then it is truly opaque what actual Sequence or Property is returned to the user by various APIs.